pocs/utils/rs232.py

```diff
@@ -59,61 +70,88 @@ def __init__(self,
         self.ser.dsrdtr = False
         self.ser.write_timeout = False
 
+        self.logger.debug('SerialData for %r created', self.name)
+
         # Properties have been set to reasonable values, ready to open the port.
-        self.ser.open()
+        try:
+            self.ser.open()
+        except serial.serialutil.SerialException as err:
+            self.logger.debug('Unable to open %r. Error: %r', self.name, err)
+            return
 
         open_delay = max(0.0, float(open_delay))
-        self.logger.debug(
-            'Serial connection set up to %r, sleeping for %r seconds', self.name, open_delay)
         if open_delay > 0.0:
+            self.logger.debug('Opened %r, sleeping for %r seconds', self.name, open_delay)
             time.sleep(open_delay)
-        self.logger.debug('SerialData created')
+        else:
+            self.logger.debug('Opened %r', self.name)
 
     @property
     def is_connected(self):
         """True if serial port is open, False otherwise."""
-        connected = False
-        if self.ser:
-            connected = self.ser.isOpen()
-
-        return connected
+        return self.ser.is_open
 
     def connect(self):
-        """Actually set up the Thread and connect to serial."""
-        # TODO(jamessynge): Determine if we need this since the serial port is opened
-        # when the instance is created.
-        self.logger.debug('Serial connect called')
-        if not self.ser.isOpen():
-            try:
-                self.ser.open()
-            except serial.serialutil.SerialException as err:
-                raise BadSerialConnection(msg=err)
-
-        if not self.ser.isOpen():
-            raise BadSerialConnection(msg="Serial connection is not open")
-
-        self.logger.debug('Serial connection established to {}'.format(
-            self.name))
-        return self.ser.isOpen()
+        """If disconnected, then connect to the serial port.
+
+        Raises:
+            BadSerialConnection if unable to open the connection.
+        """
+        if self.is_connected:
+            return
```

Comment:
Does it generate too much noise if we have a log message here saying connection was already open?

Reply:
Added appropriate message.